File: Beakjoon/_etc/C#/1748 수 이어 쓰기1.cs
using System;
using static System.Console;

class Program
{
    
    static int findDigit(int number) {
        int result = 0;
        
        while(number > 0) {
            number = number / 10;
            result += 1;
        }
        return result;
    }
    
    static void Main() {
        long answer = 0;
        string s = ReadLine();
        int n = int.Parse(s);
        
        int frontNum = s[0] - '0';
        int elseNum = 0;
        if(s.Length > 1) {
            elseNum = int.Parse(s.Substring(1));
        }
        int d = findDigit(n); int value = 0;
        for(int i = 0 ; i < d; i++) {
            if(i==0) continue;
            else {
                value += (i * (int)Math.Pow(10, i-1)) * 9;
            }
        }
        
        answer = value + (d * (int)Math.Pow(10, d-1)) * (frontNum-1) + ((elseNum+1) * d);
        
        WriteLine(answer);
    }
}

File: mvc1/_etc/Java/1748 수 이어 쓰기 1.java
import java.util.*;

public class Main {
    
    public static int findDigit(int number) {
        int result = 0;
        while(number > 0) {
            number /= 10;
            result += 1;
        }
        return result;
    }
    
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        long answer = 0;
        int n = sc.nextInt();
        int d = findDigit(n);
        int temp = n; int value = 0;
        while(temp >= 10) {
            temp /= 10;
        }
        int f = temp;
        int e = n % (int)Math.pow(10, d-1);
        for(int i = 1 ; i < d; i++) {
            value += (i * (int)Math.pow(10, i-1)) * 9;
        }
        answer = value + ((d * (int)Math.pow(10, d-1)) * (f-1)) + (e+1)*d;
        System.out.println(answer);
    }
}
